internal/service/projectipaccesslist/model_project_ip_access_list_test.go

```diff
@@ -0,0 +1,215 @@
+package projectipaccesslist_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework-timeouts/resource/timeouts"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/service/projectipaccesslist"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/atlas-sdk/v20231115002/admin"
+)
+
+var (
+	AWSSecurityGroup = "AWSSecurityGroup"
+	CIDRBlock        = "CIDRBlock"
+	IPAddress        = "IPAddress"
+	Comment          = "Comment"
+	GroupID          = "GroupID"
+	IPAddressID      = conversion.EncodeStateID(map[string]string{
+		"entry":      IPAddress,
+		"project_id": GroupID,
+	})
+	CidrBlockID = conversion.EncodeStateID(map[string]string{
+		"entry":      CIDRBlock,
+		"project_id": GroupID,
+	})
+	AwsSecurityGroupID = conversion.EncodeStateID(map[string]string{
+		"entry":      AWSSecurityGroup,
+		"project_id": GroupID,
+	})
+)
+
+func TestNewMongoDBProjectIPAccessList(t *testing.T) {
+	testCases := []struct {
+		tfModel        *projectipaccesslist.TfProjectIPAccessListModel
+		expectedResult *[]admin.NetworkPermissionEntry
+		name           string
+	}{
```

Comment:
i suppose it's ok to follow the same structure even if there is only one test

Reply:
Kept it with this format for consistency even though it's not needed

Reply:
think this is useful as it simplifies when a new test is added